src/prommis/uky/costing/tests/test_ree_costing.py

```diff
@@ -4434,3 +4465,98 @@ def test_REE_costing_recovery_Nonewithtransportcost():
+    m.fs.CS_jaw_crusher.power.fix()
+    m.fs.CS_jaw_crusher.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": CS_jaw_crusher_accounts,
+            "scaled_param": m.fs.CS_jaw_crusher.power,
+            "source": 1,
+        },
+    )
+
+    m.fs.feed_input = pyo.Var(initialize=500, units=pyunits.ton / pyunits.hr)
+    m.fs.feed_input.fix()
+
+    m.fs.water = pyo.Var(m.fs.time, initialize=1000, units=pyunits.gallon / pyunits.hr)
+    m.fs.water.fix()
+
+    m.fs.costing.build_process_costs(
+        fixed_OM=True,
+        pure_product_output_rates={
+            "Sc2O3": 1.9 * pyunits.kg / pyunits.hr,
+        },
+        mixed_product_output_rates={
+            "Sc2O3": 0.00143 * pyunits.kg / pyunits.hr,
+        },
+        variable_OM=True,
+        feed_input=m.fs.feed_input,
+        resources=[
+            "water",
+        ],
+        rates=[
+            m.fs.water,
+        ],
+        calculate_NPV=True,
+        discount_percentage=10,  # percent
+        plant_lifetime=20,  # years
+    )
+
+    dt = DiagnosticsToolbox(model=m, variable_bounds_violation_tolerance=1e-4)
+    dt.assert_no_structural_warnings()
+
+    QGESSCostingData.costing_initialization(m.fs.costing)
+    QGESSCostingData.initialize_fixed_OM_costs(m.fs.costing)
+    QGESSCostingData.initialize_variable_OM_costs(m.fs.costing)
+    solver = get_solver()
+    results = solver.solve(m, tee=True)
+    assert_optimal_termination(results)
+    dt.assert_no_numerical_warnings()
+
+    # check that some objects builts as expected
+    assert hasattr(m.fs.costing, "pv_capital_cost")
+    assert hasattr(m.fs.costing, "pv_operating_cost")
+    assert hasattr(m.fs.costing, "pv_revenue")
+    assert hasattr(m.fs.costing, "pv_royalties")
+    assert hasattr(m.fs.costing, "loan_debt")
+    assert hasattr(m.fs.costing, "loan_annual_payment")
+    assert hasattr(m.fs.costing, "pv_loan_interest")
+    assert hasattr(m.fs.costing, "pv_capital_depreciation")
+    assert hasattr(m.fs.costing, "npv")
+    assert hasattr(m.fs.costing, "discount_percentage")
+    assert hasattr(m.fs.costing, "plant_lifetime")
+    assert hasattr(m.fs.costing, "capital_expenditure_percentages")
+    assert hasattr(m.fs.costing, "capital_escalation_percentage")
+    assert hasattr(m.fs.costing, "operating_inflation_percentage")
+    assert hasattr(m.fs.costing, "revenue_inflation_percentage")
+    assert hasattr(m.fs.costing, "royalty_charge_percentage_of_revenue")
+    assert hasattr(m.fs.costing, "debt_percentage_of_CAPEX")
+    assert hasattr(m.fs.costing, "loan_interest_percentage")
+    assert hasattr(m.fs.costing, "loan_repayment_period")
+    assert hasattr(m.fs.costing, "capital_depreciation_declining_balance_percentage")
+
+    # check some NPV results
+    assert m.fs.costing.pv_capital_cost.value == pytest.approx(-14.33727, rel=1e-4)
```

Comment:
Whilst this verifies the results of the calculations, this does very little to test that the variables and constraint were built as expected. I.e., this is a `component` or `integration` test, but you have no `unit` testing of the code.